include/dgl/runtime/ndarray.h

```diff
@@ -174,6 +174,13 @@ class NDArray {
    * @brief Return a new array with a copy of the content.
    */
   inline NDArray Clone() const;
+
+  /**
+   * @brief Return a copy of the NDArray in pinned (page-locked) memory.
+   * @note This is an out-place method, and the current context has to be
+   * kDGLCPU. Otherwise, it will throw an error.
+   */
+  inline NDArray PinMemory();
```

Comment:
`PinMemory_` is DGL in-house pinning strategy, which only uses `cudaHostRegister`. 
`PinMemory` utilizes PyTorch's pinning memory pool (cached), to avoid overhead of frequently pinning/unpin memory.